### ISSUE TITLE
Does Bouncer support SQL Server?

### ISSUE THREAD

**Issue by MartinHughes-BPC** [JosephSilber/bouncer#541]
All the migrations ran fine, but when I attempt to add a user to a role I get an error:

Illuminate/Database/QueryException with message 'SQLSTATE[42S02]: [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid object name 'assigned_roles'. (SQL: select * from [assigned_roles] where [role_id] in (1) and [entity_id
] in (1149) and [entity_type] = App/User)'

From the SQL, it appears to be that App/User isn't enclosed in quotation marks that's causing this. So just wondering if Bouncer doesn't support MS SQL?

**Comment by JosephSilber** [comment]
Bouncer uses Laravel's database driver, so it should work with MS SQL Server too.

**Comment by MartinHughes-BPC** [comment]
After several months I revisited this to see if I could get it to work - and managed to solve it! This is a bit of an odd one, and may only be specific to me, but I'll update here just in case anyone finds this useful in future.

In my project I have two databases. The first (bob) is owned by my project and is where the users tables and all other Laravel stuff is held. The second (larry) is a third-party database. Although both dbs exist on the same server, I defined two connections in Laravel to make sure I didn't inadvertently create migrations against larry.

However I must have been having a funny five minutes, because in the User model I'd defined $connection = 'larry' and $table = 'bob.dbo.users' - ie the model was connecting to the right table but through the wrong connection. I changed the connection in the User model to 'bob' and hey-presto! Bouncer works with no more SQL errors.

My guess is that the different connections and the fact that the User model was using a different one to the Bouncer provided models and tables was causing the generated SQL to be malformed somehow.

Anyway, I'm going to spend the rest of the day playing around with Bouncer to see what he can do :)
